### DISCUSSION TITLE
How to use definePage's props property?

### DISCUSSION THREAD

**Discussion by a1mersnow** [posva/unplugin-vue-router#513] [Questions and Help] [answered]
Before, I write:
```ts
const props = defineProps<{
  id: number
  type?: AssetType
}>()

definePage({
  props: route => ({
    id: +route.params.id,
    type: route.params.type,
  }),
  alias: '/ad/:id/:type?',
})
```

But after upgrading to 0.10.*, there is an error `Property 'id' does not exist on type '...'`.

Currently, I use `definePage` like this:
```ts
const props = defineProps<{
  id: number
  type?: AssetType
}>()

definePage({
  props: route => ({
    id: 'id' in route.params ? +route.params.id : undefined,
    type: 'type' in route.params ? route.params.type : undefined,
  }),
  alias: '/ad/:id/:type?',
})
```

Is there any good practice?

**Comment by posva** [comment, marked as answer]
For the moment you [can cast the route](https://uvr.esm.is/guide/typescript.html#extra-types) with `as RouteLocationNormalizedLoaded<'/users/[id]'>`. Maybe in the future, `definePage` can be imported from a local `./$types` but I think it will be simply easier to directly use `useRoute()`, which is easier to type, because in the future it will also allow other types than string and string[]
